src/pages/AddProposal/AddProposal.component.tsx

```diff
@@ -41,6 +45,17 @@ export const AddProposal = ({
     name: AddProposalInputs.Values,
   });
 
+  const dispatch = useDispatch();
+
+  const govOption = useWatch({
+    name: AddProposalInputs.SendProposalContract,
+    control,
+  });
+
+  useEffect(() => {
+    dispatch(proposalsActions.setGovernor(govOption));
```

Comment:
Please move dispatching action to container.

We agreed with the team to have all blockchain and redux interactions in the container, so that component is only dependent  on it's internal state(this way it would be cleaner and easier to test)